Arctic Claw/ Polar Bear Set

KingPandaYumYum opened the pull request:
![Screenshot 2025-03-12 065150](https://github.com/user-attachments/assets/88d091c0-16b9-4905-935e-7333708a3ebb)

![Screenshot 2025-03-12 065330](https://github.com/user-attachments/assets/3d311eb6-f352-4a21-988c-bb5ec06bc759)

![Screenshot 2025-03-12 065411](https://github.com/user-attachments/assets/3d12cad6-4e35-456f-bee4-bb616a547624)

![Screenshot 2025-03-12 065451](https://github.com/user-attachments/assets/5b0a0d8b-ee1d-482d-8844-fc831e218c71)

![Screenshot 2025-03-12 065525](https://github.com/user-attachments/assets/ea37414c-4c63-4a8d-9995-60c75696de8a)

![Screenshot 2025-03-12 065614](https://github.com/user-attachments/assets/c340ce64-47a4-40ff-9bf4-6853a52b75b6)

![Screenshot 2025-03-12 070905](https://github.com/user-attachments/assets/0dd327b4-0c4d-455e-a539-0739564de317)

https://github.com/user-attachments/assets/5bff80b9-4361-4acb-88f2-25b6cc550071

Full Cow King Leathers set applies a state that transforms you into the cow king. The set also now has a +10 Tectonic Slam with 2 pieces equipped.

![image](https://github.com/user-attachments/assets/9e3ad5ac-0866-4bed-b7f9-ff7e467f7998)

Delegus commented on data/global/excel/states.txt:
```diff
@@ -244,3 +244,5 @@ Armor_Green	241																																																						tors	cgrn
 Armor_Yellow	242																																																						tors	lyel																	0
 Armor_Purple	243																																																						tors	dpur																	0
 Armor_Orange	244																																																						tors	oran																	0
+polarbear	245	3			0					1	1		1	1																											1																10	50	1	1	1	druid_morph	druid_morph	1	428								0
```
Transform Column on both states to 1; light rgb to 255 for white, or whatever for the color you want the light radius to be.

Delegus commented on data/global/excel/skilldesc.txt:
```diff
@@ -235,3 +235,5 @@ cubeloot	0	0	0	0	2		SkillNameCubeLoot	SkillDescCubeLoot	SkillDescCubeLoot	SkillN
 mark of the bear	0	0	0	0	2		SkillnameMetamorphosisBear	SkillnameMetamorphosisBear	SkillnameMetamorphosisBear	SkillnameMetamorphosisBear																	74	ModStr2uPercent		par2		74	ModStr6l		par1		12	StrSkill0		len																																																																													SkillItemMetamorphosisBear	3	0
 mark of the wolf	0	0	0	0	2		SkillnameMetamorphosisWolf	SkillnameMetamorphosisWolf	SkillnameMetamorphosisWolf	SkillnameMetamorphosisWolf																	74	ModStr2g		par2		74	ModStr4c		par1		12	StrSkill0		len																																																																													SkillItemMetamorphosisWolf	3	0
 weight system	0	0	0	0	0		charmweight1	charmweight1	charmweight1	charmweight1																																																																																																													0
+polar claws	2	4	2	2	40		Skillname420	Skillname420	Skillname420	Skillname420	5												1				75	StrSkill6		enma	exma	74	StrSkill22		toht																						36	StrSkill3		usmc	256																					40	Sksyn	Skillname240	2		76	Firedplev	Skillname226	par8		76	Firedplev	Skillname230	par8																								0
```
Columns I and J you want SkillDesc420 and SkillDesco421 and have the descriptions of the skills on them in skills.json
Would probably include that artic claws synergizes with hurricane and artic blast in the one for it somehow.

KingPandaYumYum replied:
working on this one right now, just need to test a few options to display the synergies

Delegus commented on data/global/excel/sets.txt:
```diff
@@ -102,3 +102,4 @@ Corruption Coils	Corruption Coils	100	regen		-50	-50
 Maadi's Paradox	Maadi's Paradox	100	res-all		50	50					skill	Bone Spear	5	5					hp		200	200													allskills		5	5	nofreeze		1	1	oskill	Energy Shield	10	10	res-mag		15	15																	0
 The Raven's Nest	The Raven's Nest	100	skill	Raven	5	5	ravensetwolf		1	1	res-all		40	40					ravensetwolf		1	1													ravendmg		1	1	oskill	amplify damage	1	1	red-dmg%		20	20	abs-mag%		10	10	cast2		30	30													0
 Tools of Vindication	Tools of Vindication	100	skill	Vengeance	3	3																													res-all		30	30																													0
+Panda's Polar Adventure	Panda's Polar Adventure	100									oskill	Polar Bear	10	10	oskill	Polar Claws	10	10																	extra-cold		50	50	oskill	Polar Bear	10	10	pierce-cold		50	50	oskill	Polar Claws	10	10	res-all		25	25													0
```
Because of the druid skill synergies the availability of these oskills to other classes isn't really that troublesome but I imagine cold sorc with otherwise cold gear and assassin wearing Skaadi's claws may still need minor testing just to make sure.

Delegus commented on data/global/excel/skills.txt:
Probably want to update the restrict field on the other bear only skills to include polarbear

Delegus commented on data/global/excel/setitems.txt:
```diff
@@ -434,3 +434,8 @@ The Raven's Feathers	430	The Raven's Nest	uea	Wyrmhide	7	60	60		blac						5	5000
 Hand of the Vindicator	431	Tools of Vindication	whm	War Hammer	7	12	20								5	5000	2	flat-dmg/lvl	8			swing1		20	20	skill	Vengeance	3	3	oskill_hide	393	1	1																					swing1		40	40																																						0
 Bulwark of the Vindicator	432	Tools of Vindication	pa9	Gilded Shield	7	59	60								5	5000	2	res-all		35	35	balance1		40	40	block1		55	55	aura	114	12	12	oskill_hide	393	1	1																	res-cold		30	30					red-dmg%		15	15	flat-dmg/lvl	8																												0
 Belt of the Vindicator	433	Tools of Vindication	zmb	Mesh Belt	7	40	40								5	5000	2	res-fire		20	20	res-ltng		20	20	res-cold		20	20	mag%		30	30																					res-fire		30	30	flat-dmg/lvl	8			res-ltng		30	30																														0
+Panda's Pelt	434	Panda's Polar Adventure	dr6	Alpha Helm	7	30	55	bwht	bwht						5	5000	2	nofreeze		1	1	oskill	Polar Bear	1	1	move3		30	30	balance2		30	30	allskills		2	3					oskill_hide	393	1	1																																																		0
```
oskill 393 1 1 only needs to be on shields and weapons; I would move the oskills and all skills to 2 or 3 piece bonuses personally (aside from 393 which is charm weight and needs to be in the normal properties list.)

KingPandaYumYum replied:
I put the oskills on the items as their own line because when you swapped the shield off to prebuff it reset the oskill and pulled it off the bar and lost the hotkey

Delegus replied:
That sounds annoying lol

KingPandaYumYum replied:
yeah the fix was to put it on an item, stopped it from doing that. might changes the value on the helm to roll in a range like the gloves

Delegus commented on data/global/excel/skills.txt:
```diff
@@ -404,3 +404,5 @@ Merc Lightning	401		lightning												merclightningbolt_m
 Merc Ice Blast	402		ice blast												merciceblast_m																																																																															sorceress_cast_cold											ice_cast_1									merciceblast_m													1	6		none												SC	SC	xx																				1							sorceress_cast_cold			6	25														1	1					1	7	12	1	1	1																						75	Freeze Length baseline	5	Freeze Length per leve ...
 Merc Glacial Spike	403		glacial spike												mercglacialspike_m											ln34 * (100 + skill('Blizzard'.blvl) * par7) / 100	ln12																																																																			sorceress_cast_cold											ice_cast_2									mercglacialspike_m													1	7		none												SC	SC	SC																				1							sorceress_cast_cold			18	25														1	1					1	7	20	1	1	1	 ...
 Merc Chilling Armor	404		chilling armor		18													chillingarmorbolt						chillingarmor		ln34+(skill('Frozen Armor'.blvl)+skill('Shiver Armor'.blvl))*par7		skill_armor_percent	ln12											hitbymissile	1																																																					sorceress_cast_cold				sorceress_shiverarmor							ice_cast_3	chillarmor_hit									chillingarmorbolt												1	5		none												S ...
+Polar Claws	405		polar claws	58	2										polar_claw_splash																																																																																																			polar_claw_splash													1	0		h2h												A1	A1	xx				1		1	1								1															18	25								2	wolf	bear	polarbear			1	1				1	1	8	4	0	1	1										0	Damage %																									25	Damage synergy									1	50	15						8	128		 ...
```
Testing without the full set the damage isn't even close to enough for hell higher than P1 and even that is a struggle, not sure I would nerf it much if you plan to do so, maybe knock some % off the synergies.

KingPandaYumYum replied:
yeah tbh I have not tested it a bunch without the full set, ill do some of that. also don't think a sorc can abuse it much but was meaning to test that (loses the helm).

Delegus commented on data/global/excel/states.txt:
```diff
@@ -244,3 +244,5 @@ Armor_Green	241																																																						tors	cgrn
 Armor_Yellow	242																																																						tors	lyel																	0
 Armor_Purple	243																																																						tors	dpur																	0
 Armor_Orange	244																																																						tors	oran																	0
+polarbear	245	3			0	1				1	1		1	1																											1																10	50	255	255	255	druid_morph	druid_morph	1	428								0
```
![image](https://github.com/user-attachments/assets/8d54e728-784b-4496-8052-22951f28baff)
Circled in red to 1

Delegus commented on data/local/lng/strings/skills.json:
```diff
@@ -36548,5 +36548,73 @@
+    {
+        "id": 56060,
+        "Key": "SkillDesc420",
+        "enUS": "Arctic Claws Receives Bonuses From: \n Arctic Blast: +25% Cold Damage Per Level \n Hurricane: +25% Cold Damage Per Level",
```
Should put this enus text in all fields. Same for the below "Transforms into a polar bear"

KingPandaYumYum replied:
yes i did forget to go back and do that

Delegus commented:
That commit was to fix the kings weapon.

Delegus commented:
And failed ffs.

Delegus reviewed (approved):
LGTM Balance feels good on the Panda Set, multiple monstats.txt and monstats2.txt fixes for house cleaning purposes. Love the cow king transformation idea.

Delegus commented on data/hd/character/enemy/cowkingelite.json:
Updated transformation json to only show 1 weapon instead of an amalgamation of all 3 variants used by the original cow king.

Delegus commented:
@Bakelt tagging you to give this a look please, as I made commits on this PR.

Bakelt reviewed (changes requested):
1. Has someone verified the lieutenants and duriel sarcophagus still spawn properly after the monstats.txt shift?
2. Panda's Sled has a prop gap
3. Is it intentional to have no local piece set bonuses?
4. Feral Rage is not normally usable in Werebear but is enabled for Polar Bear. Is this intentional? Descriptions still state werewolf only
5. Rabies is not normally usable in Werebear but has been enabled for Werebear and Polar Bear. Is this intentional? Descriptions still state werewolf only
6. Why is the synergies for Arctic Claws hardcoded?

Delegus commented:
> 1. Has someone verified the lieutenants and duriel sarcophagus still spawn properly after the monstats.txt shift?
> 2. Panda's Sled has a prop gap
> 3. Is it intentional to have no local piece set bonuses?
> 4. Feral Rage is not normally usable in Werebear but is enabled for Polar Bear. Is this intentional? Descriptions still state werewolf only
> 5. Rabies is not normally usable in Werebear but has been enabled for Werebear and Polar Bear. Is this intentional? Descriptions still state werewolf only
> 6. Why is the synergies for Arctic Claws hardcoded?

1. This was my change and yes I went around and tested them all in all 3 difficulties.

KingPandaYumYum commented:
> 5\. Rabies is not normally usable in Werebear but has been enabled for Werebear and Polar Bear. Is this intentional? Descriptions still state werewolf only

1. addressed 
2. it is intentional unless I decide to add more to the set.
3. feral rage isnt activated for werebear but is for polar. dont know if we wanna add that to the description for a set might lead to confusion.
4. Just updated rabies to include werebear in the skill desc
5. might have to elaborate on this a bit more

let me know if you guys have anymore feedback/changes.

Bakelt commented:
You have a single text to cover all the synergy text, the damage type, amount per level, and skills it synergizes with. 

Every other skill uses generic strings that are called with parameters.

A generic synergy text which is passed a string that corresponds to the skill name.

A corresponding string for the elemental type which is passed a string for the skill that provides the synergy and an integer which is provided by a specific par#. 

Right now, if you change the synergy value or synergy skills, it is all hardcoded in the description and can easily be missed in an update

KingPandaYumYum commented:
> You have a single text to cover all the synergy text, the damage type, amount per level, and skills it synergizes with.
> 
> Every other skill uses generic strings that are called with parameters.
> 
> A generic synergy text which is passed a string that corresponds to the skill name.
> 
> A corresponding string for the elemental type which is passed a string for the skill that provides the synergy and an integer which is provided by a specific par#.
> 
> Right now, if you change the synergy value or synergy skills, it is all hardcoded in the description and can easily be missed in an update


Trying my best to understand, messed with this bit some.

![image](https://github.com/user-attachments/assets/e88be27d-70a4-46c3-a68c-4bfc5298e041)

So I might not understand what exactly you are getting at (im a little retarded). These skills are not in the skill tree where synergies usually are shown. When you go to hot key the skill or look at the skill on the bar, synergies are not usually available in that string. My work around was to put them in the desc string. now im completely game to do it differently, I just might need my hand held.

I appreciate taking any time you put into this, and sorry for any ignorance lol

Bakelt commented:
> > You have a single text to cover all the synergy text, the damage type, amount per level, and skills it synergizes with.
> > Every other skill uses generic strings that are called with parameters.
> > A generic synergy text which is passed a string that corresponds to the skill name.
> > A corresponding string for the elemental type which is passed a string for the skill that provides the synergy and an integer which is provided by a specific par#.
> > Right now, if you change the synergy value or synergy skills, it is all hardcoded in the description and can easily be missed in an update
> 
> Trying my best to understand, messed with this bit some.
> 
> ![image](https://private-user-images.githubusercontent.com/196268445/432084115-e88be27d-70a4-46c3-a68c-4bfc5298e041.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQyNTA2NDUsIm5iZiI6MTc0NDI1MDM0NSwicGF0aCI6Ii8xOTYyNjg0NDUvNDMyMDg0MTE1LWU4OGJlMjdkLTcwYTQtNDZjMy1hNjhjLTRiZmM1Mjk4ZTA0MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMFQwMTU5MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMTJhM2YzNjlmNzVkZGIxZjc2ZDMyMjQ5YmZjNmE4ZWMyOWE5MWM1N2M4MmQ3OGQzOWZhODQ1MzZhNDg2MzI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.G4A1hU48v-qInFWxS-m8dcyVgnpYpuOW0CU8x4tkq2c)
> 
> So I might not understand what exactly you are getting at (im a little retarded). These skills are not in the skill tree where synergies usually are shown. When you go to hot key the skill or look at the skill on the bar, synergies are not usually available in that string. My work around was to put them in the desc string. now im completely game to do it differently, I just might need my hand held.
> 
> I appreciate taking any time you put into this, and sorry for any ignorance lol

Ah, so its a work around to put it on the tooltip. That makes more sense.

Bakelt reviewed (approved):
Comments and sled prop issue have been addressed. LGTM
